### PR TITLE
Vue migration: settings

### LINKED ISSUE #642: disable allow_user_to_change_display_name, admin can't change name in user panel
### Steps to reproduce
1. allow_user_to_change_display_name=false in config.php
2. user panel: changing name, doesnt save
### Expected behaviour

admin should be able to change user name
### Actual behaviour

see point 2. 
### Server configuration

**Operating system**: OVH shared hosting

**Web server:**  OVH shared hosting

**Database:** mysql

**PHP version:** 5.6

**Nextcloud version:** (see Nextcloud admin page) 9.0.53

**Updated from an older Nextcloud/ownCloud or fresh install:** new install

**Where did you install Nextcloud from:** nextcloud

### LINKED ISSUE #4320: Quota in UI is not correct
### Steps to reproduce
1.  Add users with different quota settings in different groups
2. Switch the view between different groups
3. The template row will be deleted and the quota shown is wrong for the users

After deleting the line 167 in "settings/js/users/users.js" the quota shown is correct.

**Nextcloud version:** 11.0.2

### LINKED ISSUE #5751: Errors when creating groups from the user list.
### Steps to reproduce
1. Login as Administrator.
2. Go to Users.
3. On one user add a new group.
4. In left panel, the new group appear twice.

### Expected behaviour
When you create group from user you should see only once in left panel.

### Actual behaviour
When you create group from user you see the new group twice.

grupoduplicado

### Server configuration.
**Nextcloud version:** 12

### LINKED ISSUE #5752: Errors on delete groups
### Steps to reproduce
1. Login as Administrator.
2. Go to Users.
3. On left panel, delete a group.
4. In user list, when you try to change the user group, you can add the deleted group. 
5. Errors appears but you see the deleted group as marked.

### Expected behaviour
When a group is deleted, the list of groups in user list should be refreshed.

### Actual behaviour
When you delete one group, the list of groups in user list doesn't refresh and admin can mark deleted group in user and get errors.

eliminar grupos

### Server configuration.
**Nextcloud version:** 12

### LINKED ISSUE #7424: Allow to set language in the user management
* as an admin it would be nice to set the language of a user in the user management

### LINKED ISSUE #8010: Duplicate admin group
I created by mistake another admin group that I cannot delete.

### Steps to reproduce
1. Create admin group in lower case.
2.
3.

### Expected behaviour
Don't create it/error

### Actual behaviour
Double admin group 

### Server configuration

**Operating system**: Debian 9.03

**Web server:** apache 2.4.25

**Database:** mariadb 10.1.26

***Nextcloud:*** 12.0.4

*** Does someone have an easy fix ?

### LINKED ISSUE #8862: Popup for last login in user management misaligned
* activate the "last login" column in the settings of the user management
* hover over the date

bildschirmfoto 2018-03-15 um 10 02 04

Maybe from @nextcloud/designers could have a look.

This is on 13.0.1

### LINKED ISSUE #8917: Disabled Users not visible for the Administrator
The same issue as in #6809 . But there somehow declared as not the same issue...
So this is a new ticket:

### Steps to reproduce
1. Log in as Admin (Full Admin not Group Admin)
2. Go to Users and Disable an existing User
3. On the left pane select the Group "Disabled"

### Expected behaviour
The Group "Disabled" should list disabled User Accounts

### Actual behaviour
The Group "Disabled" doesn't list the disabled Users. The amount (number) of disabled Users is visible.

### Server configuration

**Operating system**: CentOS Linux release 7.4.1708 (Core)

**Web server:** Apache/2.4.6 (CentOS)

**Database:** 5.5.56-MariaDB MariaDB Server

**PHP version:** 7.1.14

**Nextcloud version:** 12.0.6

**Updated from an older Nextcloud/ownCloud or fresh install:** oC8.0 --> oC8.1 --> oC8.2 --> oC9.0 --> nC9.0 --> nC10.0 --> nC11.0 --> nC12.0 --> nC12.0.6

**Where did you install Nextcloud from:** 

**Signing status:** No errors have been found.

**List of activated apps:**
Enabled:
  - activity: 2.5.2
  - admin_audit: 1.2.0
  - bruteforcesettings: 1.0.3
  - comments: 1.2.0
  - dav: 1.3.1
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_pdfviewer: 1.1.2
  - files_sharing: 1.4.0
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - files_videoplayer: 1.1.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - password_policy: 1.2.2
  - provisioning_api: 1.2.0
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - twofactor_backupcodes: 1.1.1
  - updatenotification: 1.2.0
  - workflowengine: 1.2.0
Disabled:
  - encryption
  - files_external
  - firstrunwizard
  - gallery
  - theming
  - user_external
  - user_ldap

**Nextcloud configuration:**
{
    "system": {
        "instanceid": "ocnibw9w9uqp",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "192.168.200.15",
            "xxx"
        ],
        "datadirectory": "\/home\/share\/data",
        "overwrite.cli.url": "http:\/\/localhost\/owncloud",
        "dbtype": "mysql",
        "version": "12.0.6.1",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_from_address": "share",
        "mail_domain": "xxx",
        "mail_smtphost": "192.168.200.14",
        "mail_smtpport": "25",
        "forcessl": true,
        "forceSSLforSubdomains": true,
        "loglevel": 2,
        "default_language": "en",
        "allow_user_to_change_display_name": false,
        "updatechecker": false,
        "check_for_working_htaccess": true,
        "logtimezone": "Europe\/Zurich",
        "maintenance": false,
        "theme": "xxx",
        "singleuser": false,
        "trashbin_retention_obligation": "30, 31",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "updater.release.channel": "production",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    },
    "apps": {
        "activity": {
            "enabled": "yes",
            "installed_version": "2.5.2",
            "types": "filesystem"
        },
        "admin_audit": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": "logging"
        },
        "backgroundjob": {
            "lastjob": "24"
        },
        "bruteforcesettings": {
            "enabled": "yes",
            "installed_version": "1.0.3",
            "types": ""
        },
        "comments": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": "logging"
        },
        "configreport": {
            "enabled": "no",
            "installed_version": "0.1.1",
            "types": "filesystem"
        },
        "core": {
            "backgroundjobs_mode": "cron",
            "global_cache_gc_lastrun": "1468834201",
            "installed.bundles": "[\"CoreBundle\"]",
            "installedat": "1435670744.6259",
            "lastcron": "1521636302",
            "lastupdateResult": "[]",
            "lastupdatedat": "1521636337",
            "moveavatarsdone": "yes",
            "oc.integritycheck.checker": "[]",
            "previewsCleanedUp": "1",
            "public_files": "files_sharing\/public.php",
            "public_gallery": "gallery\/public.php",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "remote_caldav": "dav\/appinfo\/v1\/caldav.php",
            "remote_calendar": "dav\/appinfo\/v1\/caldav.php",
            "remote_carddav": "dav\/appinfo\/v1\/carddav.php",
            "remote_contacts": "dav\/appinfo\/v1\/carddav.php",
            "remote_dav": "dav\/appinfo\/v2\/remote.php",
            "remote_files": "dav\/appinfo\/v1\/webdav.php",
            "remote_webdav": "dav\/appinfo\/v1\/webdav.php",
            "repairlegacystoragesdone": "yes",
            "scss.variables": "bb083d930fd63c4c89c5b4b3219dfd46",
            "shareapi_allow_mail_notification": "no",
            "shareapi_allow_public_notification": "no",
            "shareapi_default_expire_date": "no",
            "shareapi_enforce_links_password": "no",
            "shareapi_exclude_groups": "no",
            "shareapi_exclude_groups_list": "[\"\"]",
            "shareapi_only_share_with_group_members": "yes",
            "umgmt_send_email": "false",
            "umgmt_show_email": "false",
            "umgmt_show_last_login": "true",
            "umgmt_show_storage_location": "false",
            "updater.secret.created": "1521632343",
            "vendor": "nextcloud"
        },
        "dav": {
            "buildCalendarSearchIndex": "yes",
            "enabled": "yes",
            "installed_version": "1.3.1",
            "types": "filesystem"
        },
        "federatedfilesharing": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": ""
        },
        "federation": {
            "autoAddServers": "0",
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": "authentication"
        },
        "files": {
            "cronjob_scan_files": "500",
            "enabled": "yes",
            "installed_version": "1.7.2",
            "types": "filesystem"
        },
        "files_locking": {
            "enabled": "no",
            "installed_version": "",
            "types": "filesystem"
        },
        "files_pdfviewer": {
            "enabled": "yes",
            "installed_version": "1.1.2",
            "ocsid": "166049",
            "types": ""
        },
        "files_sharing": {
            "enabled": "yes",
            "incoming_server2server_share_enabled": "no",
            "installed_version": "1.4.0",
            "lookupServerUploadEnabled": "no",
            "outgoing_server2server_share_enabled": "no",
            "types": "filesystem"
        },
        "files_texteditor": {
            "enabled": "yes",
            "installed_version": "2.4.1",
            "ocsid": "166051",
            "types": ""
        },
        "files_trashbin": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": "filesystem"
        },
        "files_versions": {
            "enabled": "yes",
            "installed_version": "1.5.0",
            "types": "filesystem"
        },
        "files_videoplayer": {
            "enabled": "yes",
            "installed_version": "1.1.0",
            "types": ""
        },
        "files_videoviewer": {
            "enabled": "no",
            "installed_version": "0.1.3",
            "ocsid": "166054",
            "types": ""
        },
        "firstrunwizard": {
            "enabled": "no",
            "installed_version": "1.1",
            "ocsid": "166055",
            "types": ""
        },
        "gallery": {
            "enabled": "no",
            "installed_version": "0.6.1",
            "types": ""
        },
        "logreader": {
            "enabled": "yes",
            "installed_version": "2.0.0",
            "levels": "00111",
            "ocsid": "170871",
            "types": ""
        },
        "lookup_server_connector": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "types": "authentication"
        },
        "nextcloud_announcements": {
            "enabled": "yes",
            "installed_version": "1.1",
            "pub_date": "Sat, 10 Dec 2016 00:00:00 +0100",
            "types": "logging"
        },
        "notifications": {
            "enabled": "yes",
            "installed_version": "2.0.0",
            "types": "logging"
        },
        "oauth2": {
            "enabled": "yes",
            "installed_version": "1.0.5",
            "types": "authentication"
        },
        "password_policy": {
            "enabled": "yes",
            "enforceNumericCharacters": "1",
            "enforceUpperLowerCase": "1",
            "installed_version": "1.2.2",
            "minLength": "12",
            "types": ""
        },
        "provisioning_api": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": "prevent_group_restriction"
        },
        "serverinfo": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": ""
        },
        "sharebymail": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": "filesystem"
        },
        "survey_client": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "last_report": "{\"id\":\"ocnibw9w9uqp\",\"items\":[[\"server\",\"version\",\"12.0.3.3\"],[\"server\",\"code\",\"other\"],[\"server\",\"enable_avatars\",\"yes\"],[\"server\",\"enable_previews\",\"yes\"],[\"server\",\"memcache.local\",\"\\\\OC\\\\Memcache\\\\APCu\"],[\"server\",\"memcache.distributed\",\"none\"],[\"server\",\"asset-pipeline.enabled\",\"no\"],[\"server\",\"filelocking.enabled\",\"yes\"],[\"server\",\"memcache.locking\",\"none\"],[\"server\",\"debug\",\"no\"],[\"server\",\"cron\",\"cron\"],[\"php\",\"version\",\"5.6.31\"],[\"php\",\"memory_limit\",536870912],[\"php\",\"max_execution_time\",3600],[\"php\",\"upload_max_filesize\",8589934592],[\"database\",\"type\",\"mysql\"],[\"database\",\"version\",\"5.5.56\"],[\"database\",\"size\",146227200],[\"apps\",169116,\"disabled\"],[\"apps\",\"activity\",\"2.5.2\"],[\"apps\",\"admin_audit\",\"1.2.0\"],[\"apps\",\"bruteforcesettings\",\"1.0.2\"],[\"apps\",\"comments\",\"1.2.0\"],[\"apps\",\"configreport\",\"disabled\"],[\"apps\",\"dav\",\"1.3.0\"],[\"apps\",\"federatedfilesharing\",\"1.2.0\"],[\"apps\",\"federation\",\"1.2.0\"],[\"apps\",\"files\",\"1.7.2\"],[\"apps\",\"files_locking\",\"disabled\"],[\"apps\",\"files_pdfviewer\",\"1.1.1\"],[\"apps\",\"files_sharing\",\"1.4.0\"],[\"apps\",\"files_texteditor\",\"2.4.1\"],[\"apps\",\"files_trashbin\",\"1.2.0\"],[\"apps\",\"files_versions\",\"1.5.0\"],[\"apps\",\"files_videoplayer\",\"1.1.0\"],[\"apps\",\"files_videoviewer\",\"disabled\"],[\"apps\",\"firstrunwizard\",\"disabled\"],[\"apps\",\"gallery\",\"disabled\"],[\"apps\",\"logreader\",\"2.0.0\"],[\"apps\",\"lookup_server_connector\",\"1.0.0\"],[\"apps\",\"nextcloud_announcements\",\"1.1\"],[\"apps\",\"notifications\",\"2.0.0\"],[\"apps\",\"oauth2\",\"1.0.5\"],[\"apps\",\"password_policy\",\"1.2.2\"],[\"apps\",\"provisioning_api\",\"1.2.0\"],[\"apps\",\"serverinfo\",\"1.2.0\"],[\"apps\",\"sharebymail\",\"1.2.0\"],[\"apps\",\"survey_client\",\"1.0.0\"],[\"apps\",\"systemtags\",\"1.2.0\"],[\"apps\",\"templateeditor\",\"disabled\"],[\"apps\",\"theming\",\"disabled\"],[\"apps\",\"twofactor_backupcodes\",\"1.1.1\"],[\"apps\",\"updatenotification\",\"1.2.0\"],[\"apps\",\"workflowengine\",\"1.2.0\"],[\"stats\",\"num_files\",70137],[\"stats\",\"num_users\",134],[\"stats\",\"num_storages\",168],[\"stats\",\"num_storages_local\",2],[\"stats\",\"num_storages_home\",146],[\"stats\",\"num_storages_other\",20],[\"stats\",\"num_comments\",2],[\"stats\",\"num_comment_markers\",1],[\"stats\",\"num_systemtags\",1],[\"stats\",\"num_systemtags_mappings\",0],[\"files_sharing\",\"num_shares\",3814],[\"files_sharing\",\"num_shares_user\",18],[\"files_sharing\",\"num_shares_groups\",4],[\"files_sharing\",\"num_shares_link\",3792],[\"files_sharing\",\"num_shares_link_no_password\",3792],[\"files_sharing\",\"num_fed_shares_sent\",0],[\"files_sharing\",\"num_fed_shares_received\",0],[\"files_sharing\",\"permissions_0_1\",\"1\"],[\"files_sharing\",\"permissions_3_1\",\"3689\"],[\"files_sharing\",\"permissions_0_15\",\"6\"],[\"files_sharing\",\"permissions_3_15\",\"103\"],[\"files_sharing\",\"permissions_0_19\",\"2\"],[\"files_sharing\",\"permissions_0_31\",\"9\"],[\"files_sharing\",\"permissions_1_31\",\"4\"],[\"encryption\",\"enabled\",\"no\"],[\"encryption\",\"default_module\",\"no\"]]}",
            "last_sent": "1507545233",
            "types": ""
        },
        "systemtags": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": "logging"
        },
        "templateeditor": {
            "enabled": "no",
            "installed_version": "0.2",
            "types": ""
        },
        "theming": {
            "enabled": "no",
            "installed_version": "1.3.0",
            "types": "logging"
        },
        "twofactor_backupcodes": {
            "enabled": "yes",
            "installed_version": "1.1.1",
            "types": ""
        },
        "updatenotification": {
            "bruteforcesettings": "1.0.3",
            "core": "13.0.1.1",
            "enabled": "yes",
            "files_pdfviewer": "1.1.2",
            "installed_version": "1.2.0",
            "types": "",
            "update_check_errors": "0"
        },
        "workflowengine": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": "filesystem"
        },
        "169116": {
            "enabled": "no"
        }
    }
}

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** IE11, Firefox 59.0.1

**Operating system:** Windows 7 64-bit

### LINKED ISSUE #9191: Change own quota as group admin
### Steps to reproduce
1. Create a user
2. Make that user a group admin
3. Go to the users settings as that user
4. Change own quota

### Expected behaviour
Should be able to change quotas of other users, but not the own.

### Actual behaviour
Able to change the own quota.

### Server configuration

**Operating system:** Ubuntu 16.04.4 LTS (GNU/Linux 4.4.0-112-generic x86_64)

**Web server:** nginx/1.10.3

**Database:** 10.0.34-MariaDB-0ubuntu0.16.04.1

**PHP version:** PHP 7.0.28-0ubuntu0.16.04.1

**Nextcloud version:** 13.0.1

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from 13.0.0, but not sure if issue was available previously

**Where did you install Nextcloud from:** Web-Installer from nextcloud.com / Upgrade Tool

**Signing status:**
<details>
<summary>Signing status</summary>
No errors have been found.
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

Enabled:
  - activity: 2.6.1
  - admin_notifications: 1.0.1
  - apporder: 0.4.1
  - bookmarks: 0.11.0
  - bruteforcesettings: 1.0.3
  - calendar: 1.6.1
  - checksum: 0.4.0
  - circles: 0.13.6
  - comments: 1.3.0
  - contacts: 2.1.3
  - dav: 1.4.6
  - encryption: 2.0.0
  - federatedfilesharing: 1.3.1
  - federation: 1.3.0
  - files: 1.8.0
  - files_accesscontrol: 1.3.0
  - files_automatedtagging: 1.3.0
  - files_downloadactivity: 1.2.0
  - files_external: 1.4.1
  - files_pdfviewer: 1.2.1
  - files_retention: 1.2.0
  - files_sharing: 1.5.0
  - files_texteditor: 2.5.1
  - files_trashbin: 1.3.0
  - files_versions: 1.6.0
  - files_videoplayer: 1.2.0
  - flowupload: 0.0.7
  - gallery: 18.0.0
  - groupfolders: 1.2.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.1.0
  - metadata: 0.6.0
  - mindmaps: 0.1.0
  - nextcloud_announcements: 1.2.0
  - notes: 2.3.2
  - notifications: 2.1.2
  - oauth2: 1.1.0
  - passman: 2.1.4
  - password_policy: 1.3.0
  - polls: 0.8.1
  - provisioning_api: 1.3.0
  - quota_warning: 1.2.0
  - rainloop: 5.0.6
  - serverinfo: 1.3.0
  - sharebymail: 1.3.0
  - spreed: 3.1.0
  - survey_client: 1.1.0
  - systemtags: 1.3.0
  - theming: 1.4.1
  - twofactor_backupcodes: 1.2.3
  - twofactor_totp: 1.4.1
  - twofactor_u2f: 1.5.1
  - updatenotification: 1.3.0
  - user_usage_report: 1.0.1
  - w2g2: 2.0.5
  - workflowengine: 1.3.0
Disabled:
  - admin_audit
  - firstrunwizard
  - user_external
  - user_ldap

</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.tkleinert.de"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/cloud.tkleinert.de",
        "dbtype": "mysql",
        "version": "13.0.1.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "updater.release.channel": "stable",
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpsecure": "tls",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "theme": "",
        "loglevel": 2
    }
}

</details>

**Are you using encryption:** yes

### Client configuration
**Browser:** Google Chrome 65.0.3325.181

**Operating system:** Windows 10 - 1709 (Build 16299.192)

### LINKED ISSUE #9289: Unexpected submit is triggered on new user form
1. Click on "Add user"
2. Add a name and password (do not submit)
3. Start editing an other users display name and press enter
4. "New user is created", wait what?

Also if you don't do `2`, it will error with "empty name"

### PR DESCRIPTION
**Pull request by skjnldsv**
Features:
- Form validation
  - New user form requires user+pass or user+email minimum
  - Validate every other fields:
    - displayname can be empty
    - password can't be empty and has the minlength requirement from the password policy if enabled
    - emails are validated as emails & can be empty
- We switched to a grid design, no more ugly table
- Hide/show columns settings are now saved in the browser, saving us some undesirable db entries
- Only using our api, no more shady legacy ajax entry points
- We only show 2 groups max, and [add an indicator](https://user-images.githubusercontent.com/14975046/37727800-03f2e09c-2d39-11e8-9ee1-d45daf531071.png) saying how many more are selected (better visual & consistency)
- Password can't be change if encryption does NOT use a master key (safety reason)
- Now using Vuejs, components are fresh and will be moved to a dedicated repo later on with proper documentation and tests
- Language edition capability

Fix #642 
Fix #7424
Fix #5752 
Fix #5751
Fix #8862
Fix #9289
Fix #9191
Fix #8917 
Fix #8010
Fix #4320

Settings
- [x] Design + base
- [x] Quota
- [x] Groups management
- [x] Subadmins management
- [x] User disablement
- [x] User deletion
- [x] Props edition (email, name...)
- [x] Server-side-ecrptn password change
- [x] Migrate to OCS api
- [x] Support password policy min length
- [ ] Search support
- [x] Group deletion

Design
- [x] Better display of overflow tags
- [x] Themed vue-multiselect

Waiting for:
- [x] #8856 
- [x] #8847 
- [x] #8864 
- [x] #8865
- [x] #8904 
- [x] #8926
- [x] #9111 
- [x] #8777
- [x] #9108 
- [x] #9296 
- [ ] #9438

Components see #7738 
- [x] Popovermenu v1
- [x] AppNavigation v1
- [x] Request api + authentification system

Bugs
- [ ] On user deletion, if admin of group(s), db key(s) is/are not removed
- [x] All user count also includes disabled but they are not shown on the list
- [ ] User old apppassword are not deleted on user deletion! :warning: 

Before merge:
- [x] Drop css fixed commits after #8777 
- [x] Drop 582aece610d0d98852ee1ddf0f8f6a48e63c14cb after #9108 is merged
- [x] Drop a361b7b038c33b8215f369b199fbcc8940c00539 after  #9111 is merged

### CODE REVIEW

File: settings/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+``` bash
+# install dependencies
+npm install
+
+# serve with hot reload at localhost:8080
+npm run dev
+
+# build for production with minification
+npm run build
```
Comment (MorrisJobke) [inline settings/README.md]: Makefile please ... see `apps/updaternotifications` 😉

Comment (skjnldsv) [reply settings/README.md]: I'll implement it once done :)

File: settings/users.php [hunk 2]
```diff
@@ -109,7 +110,7 @@
 $disabledUsers = $isLDAPUsed ? 0 : $userManager->countDisabledUsers();
 $disabledUsersGroup = [
 	'id' => '_disabledUsers',
-	'name' => '_disabledUsers',
+	'name' => 'Disabled users',
```
Comment (nickvergessen) [inline settings/users.php]: translate?!

**Comment by nickvergessen** [comment]
- [ ] Loading icon/state for the items-list

File: settings/src/components/appNavigation/navigationItem.vue [hunk 3]
```diff
@@ -92,18 +92,18 @@ export default {
 		}
 	},
 	methods: {
-		showMenu () {
+		showMen() {
```
Comment (nickvergessen) [inline settings/src/components/appNavigation/navigationItem.vue]: ?

Comment (skjnldsv) [reply settings/src/components/appNavigation/navigationItem.vue]: Error indeed

**Comment by skjnldsv** [comment]
Okay everyone, this is ready for review!
This is unfortunately a pretty big PR, because I cleaned a lot of old legacy stuff. Please don't be afraid! <3

@nextcloud/designers 

What to test:
- Addition/removal of users
- Addition/removal of groups
- Quota change
- Default quota change
- Password/displayname/email changes
- Groups/subadmins edition
- Custom quotas set/unset
- Group change on the app navigation
- Load the page with http://xxxx/index.php/xxx and without `index.php`
- Language change on users
- Password policy min length requirements

**Comment by juliushaertl** [comment]
Really nice so far, just a few things from having a first try, i'll have a more detailed look tomorrow:

- When index.php is required for the URL the router doesn't seem to add it when clicking on groups in the navigation sidebar
- Clicking the 3-dots menu in the user list for the second time doesn't close the popover again
- The delete button for groups should be hidden by default in my opinion and just be shown on hover (or maybe as it was before just if you selected the group)
- A bit more space between the columns would be good for when inputs are besides plain text 
image
- The group multiselect input should have a pointer cursor to indicate that it is clickable
- Text fields and multiselect fields should have the same height
image
- When clicking the new user button, the first input field should be focused

**Comment by skjnldsv** [comment]
> The delete button for groups should be hidden by default in my opinion and just be shown on hover (or maybe as it was before just if you selected the group)

I'm not sure on this. I agree that it looks strange, but it's also a bad UX for users, like you don't know where you can delete the group. We could try something else tough, but I don't have any idea on this atm :(

Thanks for your review, I'll fix this asap :heart:

File: settings/src/router.js [hunk 4]
```diff
@@ -15,6 +15,9 @@ Vue.use(Router);
 
 export default new Router({
 	mode: 'history',
+	// if index.php is in the url AND we got this far, then it's working:
+	// let's keep using index.php in the url
+	base: window.location.pathname.indexOf('index.php') > 0 ? '/index.php/' : '/',
```
Comment (juliushaertl) [inline settings/src/router.js]: I think that would still break when installed into a subdirectory. Can we just use `OC.generateUrl('')`?

Comment (skjnldsv) [reply settings/src/router.js]: Good call. I need to determine the base url yes :)

**Review by juliusknorr — CHANGES_REQUESTED** [review]

**Review by juliusknorr — DISMISSED** [review]
Looks good code wise, just two bugs with groups management found.

**Comment by juliushaertl** [comment]
Two other issues with groups:
- [ ] Add a user while you are viewing a group list -> User is added without a group, imo we should prefill the group list with the current group there
- [ ] If you unassign a user from the group you are currently viewing it should be removed from the list

File: lib/private/L10N/Factory.php [hunk 5]
```diff
@@ -441,4 +446,74 @@ public function createPluralFunction($string) {
...
 		}
 	}
+
+	/**
+	 * returns the common language and other languages in an
+	 * associative array
+	 *
+	 * @return array
+	 */
+	public function getLanguages() {
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply lib/private/L10N/Factory.php]: For the sanity of reviewers, I guess tests for this will come later ^^

**Comment by MorrisJobke** [comment]
> We switched to a grid design, no more ugly table

How does it look in IE?

**Comment by skjnldsv** [comment]
> How does it look in IE?

Descent for ie11!
Some center align issues, but nothing terrible. I added a config for ie polyfills. 
In the end we should really drop ie11!

**Comment by michag86** [comment]
Hi,
thanks for the work on the user management!

I have tested it and found some issues:
- [ ] Groups with long names are cut off in the group selection of the user.
- [ ] It would be nice if the results are sortable
- [ ] Problems for subadmins (/ocs/v2.php/apps/provisioning_api/api/v1/config/apps/password_policy/minLength) results in error 997 "Logged in user must be an admin"
- [ ] If subadmin trys to create a user without selecting a group, the spinner is spinning for ever. 
- [ ] Subadmin can see the subadmin permissions and can try to change subadmin permissions. Maybe this should be hidden, hideable or read only for subadmins?
- [ ] User count for everyone (as subadmin) is not the count of the subadmin scope. Instead it represents all users.
- [ ] Subadmins can try to create a group and the spinner is spinning for ever.

**Comment by skjnldsv** [comment]
> - It would be nice if the results are sortable

Maybe in a future pr :)

> - Problems for subadmins (/ocs/v2.php/apps/provisioning_api/api/v1/config/apps/password_policy/minLength) results in error 997 "Logged in user must be an admin"

Have to be fixed in the policy app: https://github.com/nextcloud/password_policy

**Comment by skjnldsv** [comment]
@nextcloud/designers bump :)

**Comment by skjnldsv** [comment]
@michag86 small push to update the width of the groups list
Also: added a tooltip to show the full list on hover: 
capture d ecran_2018-05-18_16-58-20

**Review by MorrisJobke — APPROVED** [review]
Tested and works a lot better than before - some minor things for Edge and IE, but this is stuff for later 👍

**Review by rullzer — APPROVED** [review]
:elephant:

**Comment by danopz** [comment]
Will the components be pushed to [nextcloud/vue-components](https://github.com/nextcloud/vue-components)?

**Comment by skjnldsv** [comment]
@danopz very soon yes :)

**Pull request merged**